File: ios/Modules/Sources/Stickers/api/StickerCatalog.swift
import Foundation

public protocol StickerCatalog: Sendable {
    var favorites: AsyncStream<[StickerPack]> { get }
    func refresh() async
    func addPack(_ id: String) async -> Bool
    func removePack(_ id: String) async -> Bool
}

public struct StickerPack: Sendable, Equatable {
    public let id: String
    public let name: String
    public let coverUrl: String
    public let stickerCount: Int
    public init(id: String, name: String, coverUrl: String, stickerCount: Int) {
        self.id = id; self.name = name; self.coverUrl = coverUrl; self.stickerCount = stickerCount
    }
}

public protocol VoiceRecorder: Sendable {
    /// Returns a session; caller calls `stop()` to retrieve bytes or `cancel()` to drop.
    func start() async throws -> VoiceSession
}

public protocol VoiceSession: Sendable {
    func stop() async -> Data
    func cancel() async
}

File: android/app/src/main/java/sol/vortexx/android/stickers/api/StickerCatalog.kt
package sol.vortexx.android.stickers.api

import kotlinx.coroutines.flow.Flow

/** Wave 16: sticker packs + saved GIFs + voice notes. */
interface StickerCatalog {
    fun favoritePacks(): Flow<List<StickerPack>>
    suspend fun addPack(packId: String): Boolean
    suspend fun removePack(packId: String): Boolean
}

data class StickerPack(
    val id: String,
    val name: String,
    val coverUrl: String,
    val stickerCount: Int,
)

interface VoiceRecorder {
    suspend fun start(): VoiceSession
}

interface VoiceSession {
    suspend fun stop(): ByteArray              // Opus-in-Ogg
    suspend fun cancel()
}
